jakarta-eclipse/src/org/jakarta/codeAction/HttpServletQuickFix.java

```diff
@@ -0,0 +1,44 @@
+package org.jakarta.codeAction;
+
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.ITypeBinding;
+import org.eclipse.jdt.internal.corext.dom.Bindings;
+import org.eclipse.jdt.internal.core.manipulation.util.BasicElementLabels;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.Diagnostic;
+
+import org.jakarta.jdt.ServletConstants;
+import org.jakarta.codeActionProposal.ChangeCorrectionProposal;
+
+public class HttpServletQuickFix implements IJavaCodeActionParticipant{
+	@Override
+	public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic,
+			IProgressMonitor monitor) throws CoreException {
+		ASTNode node = context.getCoveredNode();
+		ITypeBinding parentType = Bindings.getBindingOfParentType(node);
+		if (parentType != null) {
+			List<CodeAction> codeActions = new ArrayList<>();
+			// Create code action
+			// interface
+			final String TITLE_MESSAGE = "Let ''{0}'' extend ''{1}''";
+			String args[] = { BasicElementLabels.getJavaElementName(parentType.getName()),
+					BasicElementLabels.getJavaElementName(ServletConstants.HTTP_SERVLET) };
+			ChangeCorrectionProposal proposal = new ChangeCorrectionProposal(parentType.getName(),
+					ServletConstants.HTTP_SERVLET, null);
```

Comment:
`ChangeCorrectionProposal` is accepting `name`, `kind`, `Change`.  The `kind` should be `CodeActionKind.QuickFix`. You cannot pass `null` in for the change parameter, as that indicates no code change needs to be made and the quick fix will not appear